src/main/java/io/stargate/sgv2/jsonapi/service/embedding/DataVectorizer.java

```diff
@@ -102,12 +106,28 @@ public Uni<Boolean> vectorize(List<JsonNode> documents) {
             .onItem()
             .transform(
                 vectorData -> {
+                  // check if we get back the same number of vectors that we asked for
+                  if (vectorData.size() != vectorizeTexts.size()) {
+                    throw EMBEDDING_PROVIDER_INVALID_RESPONSE.toApiException(
+                        "Embedding provider '%s' cannot return the correct number of vectors. Expect: '%s'. Actual: '%s'",
+                        collectionSettings.vectorConfig().vectorizeConfig().provider(),
+                        vectorizeTexts.size(),
+                        vectorData.size());
+                  }
                   for (int vectorPosition = 0;
                       vectorPosition < vectorData.size();
                       vectorPosition++) {
                     int position = vectorizeMap.get(vectorPosition);
                     JsonNode document = documents.get(position);
                     float[] vector = vectorData.get(vectorPosition);
+                    // check if all vectors have the expected size
+                    if (vector.length != collectionSettings.vectorConfig().vectorSize()) {
+                      throw EMBEDDING_PROVIDER_INVALID_RESPONSE.toApiException(
+                          "Embedding provider '%s' cannot return correct vector length. Expect: '%s'. Actual: '%s'",
```

Comment:
Can this be rephrased as `Embedding provider '%s' did not return expected embedding length`?

Reply:
We don't need to mention the actual length and the expected length?